app/models/checkbox_facet.rb

```diff
@@ -20,6 +20,10 @@ def has_filters?
     is_checked?
   end
 
+  def has_value?
```

Comment:
You could also have a ```query_hash``` method that returns either an empty hash if none exists or a hash with key->value pair.

In the method ```FinderPresenter#alert_query_string``` you can then simply combine them, so that all logic is contained in the facet models.

What do you think?

Reply:
I did think about that, in the hope that I could then get rid of the `has_filters?` method too but unfortunately that's used elsewhere. I hemmed and hawed a bit and in the end decided that I felt slightly happier about `has_value` as being a clear method that just does one thing, (although so would `query_hash`) so.... I dunno, it's a bit six of one really. If you have strong feelings, I'll bow to your superior tallness